### PR TITLE
Prusti dev guide

### PR DESCRIPTION
**Pull request by Aurel300**
[**Link to rendered preview**](http://media.thenet.sk/aca/prusti-dev-guide/) (as of 6768c71)

(Sorry about the notification noise; this is #124 but now applied onto the orphan `gh-pages-src` branch.)

### CODE REVIEW

File: dev-guide/src/encoding/types-snap.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+function hello(a: Int): Int
+  requires true
+  ensures result == mirror_hello(a)
+{
+  !(a < 3) ? (!(a > 10) ? 10 : a - 2) : a + 2
+}
+```
+
+TODO
```
Comment (Aurel300) [inline dev-guide/src/encoding/types-snap.md]: @cmatheja You might want to look if [what I wrote for snapshots so far](http://media.thenet.sk/aca/prusti-dev-guide/encoding/types-snap.html) is reasonable and expand on this mirror section.

File: dev-guide/src/intro.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Introduction
+
+This is the developer guide for [Prusti](https://github.com/viperproject/prusti-dev/), intended to make Prusti more approachable for new contributors.
+
```
Comment (cmatheja) [inline dev-guide/src/intro.md:4]: Perhaps highlight *contributors* and add the following sentence:

For installation instructions and a tutorial on using Prusti, please confer the [user guide](URL).

**Comment by cmatheja** [comment]
I think the current state of the dev guide can be merged. We can expand a bit on individual items, e.g. the encoding of loops, later or whenever there is a significant change.

**Comment by Aurel300** [comment]
Here we are: https://viperproject.github.io/prusti-dev/dev-guide/

**Pull request merged**